matrix-client.el

```diff
@@ -267,22 +281,27 @@ Otherwise, use the room name or alias."
 If [`matrix-client-use-auth-source'] is non-nil, attempt to log
 in using data from auth-source.  Otherwise, prompt for username
 and password."
-  (let* ((auth-source-creation-prompts (a-list 'username "Matrix identity: "
-                                               'secret "Matrix password for %u (homeserver: %h): "))
-         (found (nth 0 (auth-source-search :max 1
-                                           :host (oref con :base-url)
-                                           :user username
-                                           :require '(:user :secret)
-                                           :create t))))
-    (when (and found
-               (matrix-login-with-password con (plist-get found :user)
-                                           (let ((secret (plist-get found :secret)))
-                                             (if (functionp secret)
-                                                 (funcall secret)
-                                               secret))))
-      (oset con :username (plist-get found :user))
-      (when-let ((save-func (plist-get found :save-function)))
-        (funcall save-func)))))
+  (let* ((username (or username (read-from-minibuffer "Username: ")))
```

Comment:
Yeah, I looked into the auth system code, and I found it a bit confusing.  Basically, I don't think we should be storing passwords at all, since Matrix has tokens.  So I wanted to store tokens with the auth system, but I'm not sure if it's even possible, because it seems to be designed to store a secret that has been given by the user before attempting connection rather than something given by the server after connecting.  It seems that to do so would require copying some of the auth system code into our package and calling underlying code directly rather than using the expected API.  I recoiled at that idea and decided to go for this simple approach instead.  If you're brave enough to dig into all that, feel free.  :)